### PR TITLE
ISSD-2694 [PRM] Create a Client Extension Spring Boot for Account Sync

### PR DESCRIPTION
**Pull request by thaynam**

### CODE REVIEW

**Comment by thaynam** [comment]
ci:test:sf

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+
+					String updatedAccount = _put(
+						partnerAccountJSONObject.toString(),
+						StringBundler.concat(
+							"/o/headless-admin-user/v1.0/accounts",
+							"/by-external-reference-code/",
+							accountExternalReferenceCode));
+
+					if (proxyAccountJSONObject.has("region")) {
+						JSONObject globalOrganizationResponseJSONObject = _get(
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:276]:
```suggestion
						JSONObject globalOrganizationJSONObject = _get(
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+							uriBuilder -> uriBuilder.path(
+								"/o/headless-admin-user/v1.0/organizations" +
+									"/by-external-reference-code/PRM-ORG-GLOBAL"
+							).build());
+
+						Long globalOrganizationId =
+							globalOrganizationResponseJSONObject.getLong("id");
+
+						JSONObject organizationsResponseJSONObject = _get(
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:285]:
```suggestion
						JSONObject childOrganizationsJSONObject = _get(
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+										"/child-organizations"
+							).queryParam(
+								"pageSize", "-1"
+							).build());
+
+						Long organizationsResponseTotalCount =
+							organizationsResponseJSONObject.getLong(
+								"totalCount");
+
+						if (organizationsResponseTotalCount > 0) {
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:298]:
```suggestion
						if (organizationsResponseTotalCount =
							organizationsResponseJSONObject.getLong(
								"totalCount") > 0) {
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+								"/o/c/partnerlevels/"
+							).queryParam(
+								"pageSize", "-1"
+							).build());
+
+						Long partnerLevelResponseTotalCoult =
+							partnerLevelsResponseJSONObject.getLong(
+								"totalCount");
+
+						if (partnerLevelResponseTotalCoult > 0) {
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:221]:
```suggestion
						if (partnerLevelsResponseJSONObject.getLong(
								"totalCount") > 0) {
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+
+										_patch(
+											accountExternalReferenceCodeJSONArray.
+												toString(),
+											StringBundler.concat(
+												"/o/headless-admin-user/v1.0",
+												"/organizations/move-accounts/",
+												organizationId, "/", regionId,
+												"/by-external-reference-code"));
+									}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java:364]:
```suggestion
												"/by-external-reference-code"));

                                                break;
									}
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Comment by thaynam** [comment]
ci:test:sf

**Comment by jairmedeiros** [comment]
ci:test:sf

**Comment by jairmedeiros** [comment]
ci:test:sf

**Comment by amosfong** [comment]
@thaynam ,

here are my changes (I didn't include your latest commit)
https://github.com/amosfong/liferay-portal/commit/57f36ea190e3cc8be78f4bc211425939c35ab308

- Sort whenever possible. Usually alphabetically
- Avoid long try/catch or if blocks
- Avoid long methods
- Avoid putting logic nested inside deep for loops
-- You would have breaks after finding your matches
-- It's not as clear
- Try to keep the short circuit (return) logic close to the top of the method
- Some variable name changes

One issue that should be changed is the oauthtoken is going to be expired at some point cause it's retrieved one time on initialization. liferay-customer-etc-cron should have an example of how to do it

We can meet to go over the changes if you have any questions. When is a good time?

**Comment by thaynam** [comment]
@amosfong thank you for the inputs! I set a meet in your calendar

**Comment by thaynam** [comment]
[resent with new changes](https://github.com/is-solutions-delivery/liferay-portal/pull/5527)

**Pull request closed without merging**